CMakeLists.txt

```diff
@@ -13,8 +28,9 @@ if(PROJECT_IS_TOP_LEVEL)
 
   option(BUILD_SAMPLES "Build sample components" ON)
 
-  set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+  option(BUILD_SHARED "Build shared libraries" OFF)
```

Comment:
This shouldn't be under if(PROJECT_IS_TOP_LEVEL);

We need the option even if we are in a subdirectory.